.github/workflows/php-js-e2e-tests.yml

```diff
@@ -1,169 +1,175 @@
+      - name: Get Composer Cache Directory
+        id: composer-cache
+        run: |
+          echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+      - uses: actions/cache@v3
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.0'
+          coverage: none
+          tools: composer
+
+      - name: Composer install
+        run: |
+          composer install
+      - name: E2E Tests (WP latest with Gutenberg plugin)
+        env:
+          WOOCOMMERCE_BLOCKS_PHASE: 3
+          GUTENBERG_EDITOR_CONTEXT: 'gutenberg'
+        run: |
+          node ./bin/wp-env-with-gutenberg.js
+          npm run wp-env start
+          npm run wp-env:config && npx cross-env NODE_CONFIG_DIR=tests/e2e/config wp-scripts test-e2e --config tests/e2e/config/jest.config.js --listTests > ~/.jest-e2e-tests
+          npx cross-env JEST_PUPPETEER_CONFIG=tests/e2e/config/jest-puppeteer.config.js cross-env NODE_CONFIG_DIR=tests/e2e/config wp-scripts test-e2e --config tests/e2e/config/jest.config.js --runInBand --runTestsByPath $( awk 'NR % 5 == ${{ matrix.part }} - 1' < ~/.jest-e2e-tests )
+
+      - name: Upload artifacts on failure
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: e2e-with-gutenberg-test-report-${{matrix.part}}
+          path: reports/e2e
+
+      - name: Archive flaky tests report
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: flaky-tests-report-${{ matrix.part }}
+          path: flaky-tests
+          if-no-files-found: ignore
+  
+  JSE2ETests:
+    name: JavaScript E2E Tests (latest)
+    strategy:
+      fail-fast: false
+      matrix:
+        part: [1, 2, 3, 4, 5]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Cache node_modules
+        id: cache-node-modules
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-modified-build-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-modified-build-${{ env.cache-name }}-
+            ${{ runner.os }}-modified-build-
+            ${{ runner.os }}-modified-
+
+      - name: Setup node version and npm cache
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+
+      - name: Install Node dependencies
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
+        run: |
+          sed -i 's/"@wordpress\/env": ".*"/"@wordpress\/env": "^4.9.0"/' package.json
```

Comment:
We need to update` @wordpress/env` which caused some jobs to fail due to a [permissions problem](https://github.com/WordPress/gutenberg/issues/22515). Fixing perms wasn't fully working for E2E so we're downgrading wp-env for this job particularly.

Reply:
I just wanted to note that I've actually fixed this issue upstream and you might not need to do this anymore!